x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/store/NativeRolesStore.java

```diff
@@ -217,6 +222,48 @@ public void getRoleDescriptors(Set<String> names, final ActionListener<RoleRetri
         }
     }
 
+    public boolean isMetadataSearchable() {
+        return securityIndex.isMigrationsVersionAtLeast(ROLE_METADATA_FLATTENED_MIGRATION_VERSION);
```

Comment:
I haven't thought about it before.
I added it, per your suggestion, although I'm not sure I prefer it, since it makes it harder to understand.
I hope maybe it avoids some unnecessary 503 to users...

Reply:
Pushed [e6bb33c](https://github.com/elastic/elasticsearch/pull/108733/commits/e6bb33c99ff3f8d084dfb5fd273801ab9d4a5150)